src/egl/drivers/dri2/platform_android.c

```diff
@@ -189,6 +191,16 @@ droid_window_dequeue_buffer(struct dri2_egl_surface *dri2_surf)
 {
    int fence_fd;
 
+   _EGLSurface *surf =  &dri2_surf->base;
+
+   if ((!surf->SwapInterval && !dri2_surf->override_max_buffer_count) ||
+      (surf->SwapInterval && dri2_surf->override_max_buffer_count)) {
+      dri2_surf->override_max_buffer_count =
+           surf->SwapInterval ? 0 : ANDROID_ASYNC_MAX_BUFFER_COUNT;
```

Comment:
I'm not sure if I understand what is going on here. Shouldn't this set the "surf->SwapInterval" and not 0 if such was set?

Reply:
Also are we making Android to use 5 buffers instead of 3 here when swapinterval is 0? Could you add some comment what is expected to happen here on different intervals?

Reply:
In fact,  The logic here is followed by the previous  gfx driver by VPG.  They increased the buffer number to 5 when  aysnc mode (swapinterval 0) and set it back to default  when async mode is disabled.  I had no idea about the reason why they had to increase the number of buffers either.

Reply:
And I am sorry, you are right, I missed something , the surf->Swapinterval should be set in droid_swap_interval

Reply:
I think we should skip setting buffers to 5 if there is no good reason to do so. It is also strange things todo because on swapInterval 0 expectation is that there is no waiting for vertical sync events (same as we would run with vblank_mode=0 and tear). Would be good to know what was their reasoning for this.

Reply:
I guess this just helps prepare buffers well in advance and not blocked by buffer queue. Tripple buffer should be fine I believe in our case as every where else that will be the expectation, double or tripple buffer.